test/unittests/commands/ingest_xmlupload/test_user_information.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+from pathlib import Path
+
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from dsp_tools.commands.ingest_xmlupload import user_information
+from dsp_tools.commands.ingest_xmlupload.user_information import IngestInformation
+
+user_information.csv_filepath = Path()
+user_information.maximum_prints = 1
+
+
+class TestIngestInformation:
+    def test_no_problems(self) -> None:
+        expected = (
+            "All media referenced in the XML file were uploaded to sipi.\n"
+            "No media was uploaded to sipi that was not referenced in the XML file."
+        )
+        assert IngestInformation([], []).all_good_msg() == expected
+
+    def test_not_all_good(self) -> None:
+        assert not IngestInformation(["unused_media"], []).all_good_msg()
+
+    def test_unused_media(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    The following media were uploaded to sipi but not referenced in the data XML file:\n"
+            "    - unused_media"
+        )
+        res_msg = IngestInformation(["unused_media"], [])._get_error_msg()
+        assert res_msg == expected
+
+    def test_not_uploaded(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    The following media were not uploaded to sipi but referenced in the data XML file:\n"
+            "    - Resource ID: 'no upload id' | Filepath: 'media path'"
+        )
+        res_msg = IngestInformation([], [("no upload id", "media path")])._get_error_msg()
+        assert res_msg == expected
+
+    def test_all_problem_with_df_msg(self) -> None:
+        expected = (
+            "The upload cannot continue as there are problems with the media referenced in the XML.\n"
+            "    Media was uploaded to Sipi which was not referenced in the XML file.\n"
+            "    The file 'UnusedMediaUploadedInSipi.csv' was saved in '.' with the filenames.\n\n"
+            "    Media was referenced in the XML file but not previously uploaded to sipi:\n"
+            "    The file 'NotUploadedFilesToSipi.csv' was saved in '.' with the resource IDs and filenames."
+        )
+        res_msg = IngestInformation(
+            [
+                "unused_media0",
+                "unused_media1",
+            ],
+            [
+                "no_up0",
+                "no_up1",
+            ],
+        )._get_error_msg()
+        assert res_msg == expected
+
+
+def test_unused_media_to_df() -> None:
+    unused_media_list = [
+        "unused_media0",
+        "unused_media1",
+    ]
+    expected = pd.DataFrame({"Media Filenames": unused_media_list})
+    res_df = IngestInformation(unused_media_list, [])._unused_media_to_df()
```

Comment:
It took me a while before I understood that you monkey-patch the module `user_information` on lines 11-12. I think that's not a good practice, because it can be confusing (hard to understand). 

An alternative could be that in `user_information.py`, you transform the module-level variables into class attributes of the class `IngestInformation`, like this:

```
class IngestInformation:
    csv_filepath = field(default=Path.home(), init=False)
```

With init=False, you can exclude it from the `__init__()`, so that you don't have to pass it as parameter every time you create an instance.

For testing purposes, you can then create an instance of IngestInformation with a custom csv_filepath.

Reply:
I understand your point, but does it make sense to create a class variable that we only want because of testing?

How would you solve the maximum print statements? If we keep the current number it would bloat the test data enormously...

Reply:
The only elegant solution I see is to turn the max print statements into a class attribute, too. 
And I don't think we should do that only for testing purposes. It's not considered good practice to have module-level variables at all. So anyway, it would be better to turn them into class attributes.

Reply:
I will do that almost as said. Because if you give inti=False then you can never hand over another value, which we want (https://docs.python.org/3/library/dataclasses.html#default-factory-functions)